### PR TITLE
Bootstrap fixups

### PR DESCRIPTION
**Pull request by meatballhat**

### CODE REVIEW

File: gimme [hunk 1]
```diff
@@ -100,15 +100,18 @@ _extract() {
...
 		local bootstrap_root=""
-		for bootstrap_root in \
+		for candidate in \
+			"${GIMME_VERSION_PREFIX}/go1.4.1.${GIMME_OS}.${GIMME_ARCH}" \
+			"${GIMME_VERSION_PREFIX}/go1.4.1.${GIMME_OS}.${GIMME_HOSTARCH}" \
 			"${GIMME_VERSION_PREFIX}/go1.4.${GIMME_OS}.${GIMME_ARCH}" \
 			"${GIMME_VERSION_PREFIX}/go1.4.${GIMME_OS}.${GIMME_HOSTARCH}" ; do
-			if [ -d "${bootstrap_root}" ] ; then
+			if [ -d "${candidate}" ] ; then
+				bootstrap_root="${candidate}"
```
Comment (tianon) [inline gimme]: This seems fine, but an alternative idea (that'll catch future possible Go 1.4 versions too) would be:

``` bash
local bootstrap_root=""
local bootstrap_candidates=( "${GIMME_VERSION_PREFIX}/go1.4"*".${GIMME_OS}.${GIMME_ARCH}" )
if [ "${#bootstrap_candidates[@]}" -gt 0 ]; then
    bootstrap_root="$bootstrap_candidates" # take the first item off the array!
fi
```

(which would avoid the loop entirely also)

Comment (meatballhat) [inline gimme]: :clap: :smiley_cat:

**Comment by tianon** [comment]
Seems plenty fair (nothing unreasonable here) :+1:

File: gimme [hunk 2]
```diff
@@ -100,15 +147,19 @@ _extract() {
...
-			"${GIMME_VERSION_PREFIX}/go1.4.${GIMME_OS}.${GIMME_HOSTARCH}" ; do
-			if [ -d "${bootstrap_root}" ] ; then
-				break
-			fi
-		done
+		local bootstrap_arch_candidates=(
+			$(ls -1d "${GIMME_VERSION_PREFIX}/go1.4"*".${GIMME_OS}.${GIMME_ARCH}" 2>/dev/null)
+		)
+		local bootstrap_hostarch_candidates=(
+			$(ls -1d "${GIMME_VERSION_PREFIX}/go1.4"*".${GIMME_OS}.${GIMME_HOSTARCH}" 2>/dev/null)
```
Comment (tianon) [inline gimme]: wait what

Comment (tianon) [inline gimme]:
``` bash
local bootstrap_candidates=(
    "${GIMME_VERSION_PREFIX}/go1.4"*".${GIMME_OS}.${GIMME_ARCH}"
    "${GIMME_VERSION_PREFIX}/go1.4"*".${GIMME_OS}.${GIMME_HOSTARCH}"
)
```

(and then you only need one half of the if below, too)

File: gimme [hunk 3]
```diff
@@ -1,9 +1,56 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
 set -e
 
 [[ ${GIMME_DEBUG} ]] && set -x
 
+GIMME_VERSION=v0.1.0-8-gb1dbfdf
```
Comment (tianon) [inline gimme:52]: This is gonna be fun. :smile:

File: gimme (hunk 2 above)
Comment (meatballhat) [inline gimme]: Yup, I tried that, and the `*` was not expanding :confounded:

File: gimme (hunk 3 above)
Comment (meatballhat) [inline gimme:52]: I admit, it's mostly for purposes of Travis support.  Having an idea of what thing did what thing is pretty important, and outputting the version of a given tool prior to usage is the current convention.  I'm not going to be strict about keeping this updated, and the version of `gimme` installed in the build env will be specified in travis-cookbooks anyway.

File: gimme (hunk 2 above)
Comment (tianon) [inline gimme]: Impossibru!

Comment (tianon) [inline gimme]:
``` console
$ a=( /tmp/* )
$ echo "$a"
/tmp/adb.log
$ echo "${a[@]}"
/tmp/adb.log /tmp/aerofsupdate0.8.805234809229850226880$$$ /tmp/apt-cacher-ng /tmp/debian /tmp/dumps /tmp/go-build519429123 /tmp/gpg-cjck1N /tmp/gpg-eVkWfz /tmp/hsperfdata_portage /tmp/hsperfdata_tianon /tmp/keyring-if3Zpp /tmp/manpages-checksZjUx /tmp/orbit-tianon /tmp/pulse-kw6OfhL9weVL /tmp/steam_icon16_accessories-text-editor.png /tmp/steam_icon16_honicon_128x128.png /tmp/vepBlB8 /tmp/vmware-root /tmp/vmware-tianon /tmp/vV0z4nB
```

Comment (meatballhat) [inline gimme]: I should mention that I couldn't get `*` to expand even with only the `GIMME_ARCH` line given, which is why I got silly with `ls -1d`.

Comment (meatballhat) [inline gimme]: I DUNNO! :smiley_cat:

Comment (tianon) [inline gimme]: I guess for smashing the two together you'll need either the loop back or `shopt -s nullglob`.

File: gimme (hunk 3 above)
Comment (tianon) [inline gimme:52]: +++ sounds reasonable

File: gimme (hunk 2 above)
Comment (meatballhat) [inline gimme]: /me attempts!

**Pull request merged**